src/cpu/flexcpu/simple_dataflow_cpu.hh

```diff
@@ -0,0 +1,496 @@
+     *  result of the memory request
+     *
+     * @return Whether or not this request has been acknowledged. If this
+     *  function returns false, the requester should not expect callback_func
+     *  to be called in the future.
+     */
+    bool requestInstruction(const RequestPtr& req,
+                            FetchCallback callback_func);
+
+    /**
+     * Event-driven means for classes to request a read access to memory. Upon
+     * the response event from the memory port, the result of completeAcc()
+     * will be passed to the callback function.
+     *
+     * @param req A memory request object, which should contain a completed
+     *  translation and therefore ready to send to memory.
+     * @param callback_func The function reference for purposes of handling the
+     *  result of the memory read.
+     *
+     * @return Whether or not this request has been acknowledged. If this
+     *  function returns false, the requester should not expect callback_func
+     *  to be called in the future.
+     */
+    bool requestMemRead(const RequestPtr& req, ThreadContext* tc,
+                        StaticInstPtr inst, std::weak_ptr<ExecContext> context,
+                        MemCallback callback_func);
+
+    /**
+     * Event-driven means for classes to request a write access to memory. Upon
+     * the response event from the memory port, the result of completeAcc()
+     * will be passed to the callback function.
+     *
+     * @param req A memory request object, which should contain a completed
+     *  translation and therefore ready to send to memory.
+     * @param callback_func The function reference for purposes of handling the
+     *  result of the memory write.
+     *
+     * @return Whether or not this request has been acknowledged. If this
+     *  function returns false, the requester should not expect callback_func
+     *  to be called in the future.
+     */
+    bool requestMemWrite(const RequestPtr& req, ThreadContext* tc,
+                         StaticInstPtr inst,
+                         std::weak_ptr<ExecContext> context, uint8_t* data,
+                         MemCallback callback_func);
+
+    /**
+     * Event-driven means for classes to request a split read access to memory.
+     * Upon the response event from the memory port, the result of
+     * completeAcc() will be passed to the callback function.
+     *
+     * @param req A memory request object, which should contain a completed
+     *  translation and therefore ready to send to memory.
+     * @param callback_func The function reference for purposes of handling the
+     *  result of the memory read.
+     *
+     * @return Whether or not this request has been acknowledged. If this
+     *  function returns false, the requester should not expect callback_func
+     *  to be called in the future.
+     */
+    bool requestSplitMemRead(const RequestPtr& main, const RequestPtr& low,
```

Comment:
Mostly because if it's across a cache line, I'm imagining it could also cross a page boundary, meaning that I'd need to track two physical addresses for potential through-memory dependencies. Since the thread is already aware, it was cleaner to just pass the two translated RequestPtrs through from the thread's side.